nemo/collections/llm/recipes/llama3_70b_16k.py

```diff
@@ -0,0 +1,59 @@
+from typing import Callable
+
+import torch
+
+from nemo.collections.llm.api import pretrain
+from nemo.collections.llm.recipes import llama3_70b
+from nemo.collections.llm.utils import Partial
+
+NAME = "llama3_70b_16k"
+
+
+def pretrain_recipe(
+    name: str, ckpt_dir: str, num_nodes: int, num_gpus_per_node: int, fn: Callable = pretrain
+) -> Partial:
+    recipe = llama3_70b.pretrain_recipe(
+        name=name, ckpt_dir=ckpt_dir, num_nodes=num_nodes, num_gpus_per_node=num_gpus_per_node, fn=fn
+    )
+
+    trainer = llama3_70b.trainer(
+        tensor_parallelism=2,
+        pipeline_parallelism=4,
+        pipeline_parallelism_type=torch.bfloat16,
+        virtual_pipeline_parallelism=5,
+        context_parallelism=2,
+        sequence_parallelism=True,
+        num_nodes=num_nodes,
+        num_gpus_per_node=num_gpus_per_node,
+    )
+    model = llama3_70b.model()
+    model.config.seq_length = 16384
+
+    recipe.model = model
+    recipe.trainer = trainer
+
+    return trainer
```

Comment:
Looks like there was a bug in my PR. We want to return `recipe` here instead of `trainer`. I'll update the llama 3 8b long context recipes. Can you update all recipes here to the correct return value?

Reply:
Sure